src/sage/combinat/growth.py

```diff
@@ -1128,13 +1128,13 @@ def __eq__(self, other):
             sage: G1 == G2
             False
         """
-        return (type(self) == type(other) and
+        return (isinstance(other, GrowthDiagram) and
```

Comment:
I don't think this is a correct replacement. For equality, the types really should be identical.